src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -139,7 +140,15 @@ public static NuGetFramework ParseFrameworkName(string frameworkName, IFramework
                         profile));
                 }
 
-                result = new NuGetFramework(platform, version, profile);
+                if (version.Major >= 5
+                    && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, framework))
+                {
+                    result = new NuGetFramework(framework, version, platform ?? string.Empty, platformVersion ?? FrameworkConstants.EmptyVersion);
```

Comment:
As stated elsewhere, is `FrameworkConstants.EmptyVersion` allowed for platform version?

Reply:
It is.